UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs

```diff
@@ -122,11 +131,53 @@ public void ApplyTraumaDamage(float tramuaDamage, TraumaticDamageTypes damageTyp
 				if (damageType == TraumaticDamageTypes.PIERCE) { currentPierceDamage += MultiplyTraumaDamage(tramuaDamage); }
 				CheckCutSize();
 			}
-			//Burn damage checks for it's own armor damage type.
+			//Burn and blunt damage checks for it's own armor damage type.
 			if (damageType == TraumaticDamageTypes.BURN)
 			{
 				TakeBurnDamage(MultiplyTraumaDamage(tramuaDamage));
 			}
+
+			if (damageType == TraumaticDamageTypes.BLUNT)
+			{
+				if (DMMath.Prob(SelfArmor.Melee * 100) == false)
+				{
+					TakeBluntDamage(tramuaDamage);
+				}
+			}
+		}
+
+		public void TakeBluntDamage(float damage)
+		{
+			foreach (ItemSlot slot in OrganStorage.GetIndexedSlots())
+			{
+				if (!slot.IsEmpty)
+				{
+					if (slot.Item.gameObject.TryGetComponent<BodyPart>(out var bodyPart))
+					{
+						if (bodyPart.CanBeBroken)
```

Comment:
same here. The idea is leaving the meat and potatoes of the code block in the outer scope.